### DISCUSSION TITLE
Be related to 23w31a and 1.20.2pre1

### DISCUSSION THREAD

**Discussion by wucy777** [gnembon/fabric-carpet#1811] [Ideas]
I want to use the Carpet mod rule instruction to restore the village transactions before snapshot 23w31a and 1.20.2pre1. Of course, it can enable some villagers' experimental gameplay to be enabled on the server or client, such as two snapshots and pre released updates of villagers who have become zombies multiple times for discounted repairs, weakened library managers, cartographers selling village maps of different formations Armor craftsman trading updates, etc. (In the future, it is predicted that tool merchants and forgers may also be forced to require diamonds to trade diamond equipment). Assuming these are truly installed, I hope to use Carpet rules to separate the discount, librarian armor maker (of course, I still hope to separate the content of trading emeralds with iron blocks into a rule instruction! After all, it is really convenient), as well as possible tool vendors and forge updates to restore to the snapshot version before 23w31a!!! Because our game team has no interest in map exploration and is interested in using many large machines to produce materials to build buildings. After all, they can rely on their own will to transform the world. My world is the only one that can do this. The experimental updates of village transactions are a significant blow to us. We hope Carpet can achieve the above functions！Machine translation may not be very accurate. If the customer service team has the above ideas, they can continue to discuss with everyone below.

**Comment by altrisi** [comment]
The machine translation made that very difficult to read, but:
I think the changes are still under an experimental toggle and not definitive? You can choose not to use them yet by not enabling their experiment.
As to whether Carpet will have rules to change that back, maybe not, but an extension may have these (or a datapack if Mojang makes trades editable with datapacks)

**Reply by wucy777** [reply]
Some people in China have found that by intercepting experimental data packets of 23w31a and 1.20.2pre1, transaction content related to villages can be modified. Of course, I prefer Carpet's more refined division of these rules (the ones I have listed), which can theoretically be implemented in the future.

**Reply by wucy777** [reply]
Now, with the 1.20.2 update, it should be possible to call experimental data packets for changes.

**Reply by wucy777** [reply]
但新数据包中铁匠可以用铁块交易绿宝石，这个改动我想单独实现
